docs/docs/visual-testing-with-storybook.md

```diff
@@ -0,0 +1,103 @@
+
+To update your Storybook config open `.storybook/config.js` and add the following before the `configure` method at the bottom of the file.
+
+```js
+// Gatsby's Link overrides:
+// Gatsby defines a global called ___loader to prevent its method calls from creating console errors we override it here
+global.___loader = {
+  enqueue: () => {},
+  hovering: () => {},
+}
+
+// Gatsby internal mocking to prevent unnecessary errors in storybook testing environment
+global.__PATH_PREFIX__ = ""
+
+// This is to utilized to override the window.___navigate method Gatsby defines and uses to report what path a Link would be taking us to if it wasn't inside a storybook
+window.___navigate = pathname => {
+  action("NavigateTo:")(pathname)
+}
+```
+
+Next we need up make some adjustments to Storybook's default `webpack` configuration so that we can tranpile Gatsby source files, as well as ensure we have all the necessary `babel` plugins we need to transpile Gatsby components.
+
+Create a new file called `webpack.config.js` in the `.storybook` folder created by the Storybook CLI. Then place the following in that file.
+
+```js
+module.exports = (baseConfig, env, defaultConfig) => {
+  // Transpile Gatsby module because Gastby includes un-transpiled ES6 code.
+  defaultConfig.module.rules[0].exclude = [/node_modules\/(?!(gatsby)\/)/]
+
+  // use installed babel-loader which is v8.0-beta (which is meant to work with @babel/core@7)
+  defaultConfig.module.rules[0].use[0].loader = require.resolve("babel-loader")
+
+  // use @babel/preset-react for JSX and env (instead of staged presets)
+  defaultConfig.module.rules[0].use[0].options.presets = [
+    require.resolve("@babel/preset-react"),
+    require.resolve("@babel/preset-env"),
+  ]
+
+  // use @babel/plugin-proposal-class-properties for class arrow functions
+  defaultConfig.module.rules[0].use[0].options.plugins = [
+    require.resolve("@babel/plugin-proposal-class-properties"),
+  ]
+
+  return defaultConfig
+}
+```
+
+Once you have this configured you should run Storybook to ensure it can start up properly and you can see the default stories installed by the CLI. To run storybook:
+
+```sh
+npm run storybook
+```
+
+Storybook CLI adds this command to your `package.json` for you so you shouldn't have to anything other than run the command. If Storybook builds successfully you should be able to navigate to `http://localhost:6006` and see the default stories supplied by the Storybook CLI.
+
+## Writing stories
+
+A full guide to writing stories is beyond the scope of this guide, but we'll take a look at creating a story.
+
+First, create the story file. Storybook looks for all files with a `.story.js` extension and loads them into Storybook for you. Generally you will want your stories near where the component is defined, however since this Gatsby, if you want stories for your pages, you will have to create those files outside of the `pages` directory.
```

Comment:
```suggestion
First, create the story file. Storybook looks for all files with a `.story.js` extension and loads them into Storybook for you. Generally you will want your stories near where the component is defined, however since this is Gatsby, if you want stories for your pages, you will have to create those files outside of the `pages` directory.
```